File: ExamPrep/LabsInAllLanguages/Lab12-Python/main.py
#1
num = 2
result = (lambda x: x*2)(num)
print(result)

#2
lis = "Helao"
val = (lambda string: string[int((len(string) -1) // 2)])(lis)
print(val)


#3
fun = 5
w = (lambda num: True if num % 2 == 1 and num >= 0 else False)(fun)
print(w)


#4
test = "Abc"
val = (lambda str: True if str[0] in ["A","E","I","O","U"] else False)(test)
print(val)





val = (lambda num1, num2: num1 if(num1 > num2) else num2)
print(val(2,1))
print(val(3,52))



num1 = [8,10,7.5]
num2 = ["Hello!", "CompSci2613", "Lab-12"]
num3 = [-15, -4, 0, 4, 23, 64, 101, 104, 123]
num4 = ["alice", "bob", "Carl", "daisy", "Earl"]
#1
print(list(map(lambda x: x*2, num1)))

#2
print(list(map(lambda str: str[int (len(str) -1) // 2], num2)))

#3
print(list(filter(lambda num: True if num % 2 == 1 and num >= 0 else False , num3)))

#4
print(list(filter(lambda str: True if str[0].upper() in ["A","E","I","O","U"] else False, num4)))

File: ExamPrep/LabsInAllLanguages/Lab12-Python/main.js
const func1 = (x) => {
    return x * 2
}

const func2 = (x) => {
    return x[Math.floor(x.length / 2)]
}

const func3 = (x) => {
    if (x % 2 === 1) {
        return true
    }
    else {
        return false
    }
}

const func4 = (x) => {
    switch(x[0]) {
        case 'A':
        case 'a':
        case 'E':
        case 'e':
        case 'I':
        case 'i':
        case 'O':
        case 'o':
        case 'U':
        case 'u':
            return true;
        default:
            return false;
    }
}


result = [8,10,7.5].map(func1)

result = ["Hello!", "CompSci2613", "Lab-12"].map(func2)

result = [-15, -4, 0, 4, 23, 64, 101, 104, 123].filter((x) => x > 0 && func3(x))

result = ["alice", "bob", "Carl", "daisy", "Earl"].filter(func4)